base/abstractarray.jl

```diff
@@ -803,9 +825,8 @@ typed_hcat(T::Type, A::AbstractArray...) = cat_t(2, T, A...)
 function hvcat(nbc::Integer, as...)
     # nbc = # of block columns
     n = length(as)
-    if mod(n,nbc) != 0
-        throw(ArgumentError("all rows must have the same number of block columns"))
-    end
+    mod(n,nbc) != 0 &&
+        throw(ArgumentError("all rows must have the same number of block columns ($nbc)"))
```

Comment:
IIUC this function simply concatenates the arrays horizontally, assuming that the number of columns of the `as` arrays is equal to `nbc`. I'm not sure how it differs from `hcat`.

Anyway, I think you could improve the error message by saying that "number of arrays `$length(as)` is not a multiple of the requested number of block columns `$nbc`."